native/src/main/java/io/ballerina/lib/data/csvdata/utils/CsvUtils.java

```diff
@@ -0,0 +1,373 @@
+                return EMPTY_LONG_ARRAY;
+            }
+            return skipLinesArray.getIntArray();
+        }
+
+        return getSkipLinesFromStringConfigValue(StringUtils.getStringValue(skipLines));
+    }
+
+    public static boolean isNullValue(Object nullValue, Object value) {
+        if (value == null) {
+            return true;
+        }
+        if (value instanceof BString bString) {
+            value = StringUtils.getStringValue(bString);
+        }
+        if (value instanceof String v) {
+            return handleStringNullValue(nullValue, v, value);
+        }
+        return false;
+    }
+
+    private static boolean handleStringNullValue(Object nullValue, String v, Object value) {
+        if ((nullValue == null) && (Constants.Values.NULL.equalsIgnoreCase(v)
+                || Constants.Values.BALLERINA_NULL.equalsIgnoreCase(v))) {
+            return true;
+        }
+        return nullValue != null && value.equals(StringUtils.getStringValue(nullValue));
+    }
+
+    public static boolean isCharContainsInLineTerminatorUserConfig(char c,
+                   Object lineTerminatorObj, boolean isCarriageTokenPresent) {
+        if (lineTerminatorObj instanceof BArray array) {
+            Object[] lineTerminators = array.getValues();
+            for (Object lineTerminator: lineTerminators) {
+                Optional<Boolean> value = handleLineTerminator(lineTerminator, c, isCarriageTokenPresent);
+                if (value.isEmpty()) {
+                    continue;
+                }
+                return value.get();
+            }
+            return false;
+        }
+
+        String lineTerminator = StringUtils.getStringValue(StringUtils.fromString(lineTerminatorObj.toString()));
+        if (c == Constants.LineTerminator.LF) {
+            if (lineTerminator != null) {
+                if (lineTerminator.equals(Constants.LineTerminator.CRLF)) {
+                    return isCarriageTokenPresent;
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static Optional<Boolean> handleLineTerminator(Object lineTerminator,
+              char c, boolean isCarriageTokenPresent) {
+        if (lineTerminator == null || c != Constants.LineTerminator.LF) {
+            return Optional.empty();
+        }
+        String lineTerminatorString = lineTerminator.toString();
+        if (isCarriageTokenPresent) {
+            if (lineTerminatorString.equals(Constants.LineTerminator.CRLF)) {
+                return Optional.of(true);
+            }
+            return Optional.empty();
+        }
+        return Optional.of(true);
+    }
+
+    public static class UnMappedValue {
+        private static UnMappedValue value = null;
+        public static UnMappedValue createUnMappedValue() {
+            if (value == null) {
+                value = new UnMappedValue();
+            }
+            return value;
+        }
+    }
```

Comment:
It is not entierly clear to me what we are trying to do here. Are we trying to lazy initialize these values. In that case I think it is better to go for simple holder class idiom.


```suggestion
    public static class UnMappedValue {
        public static UnMappedValue value = createUnMappedValue();
        private static UnMappedValue createUnMappedValue() {
            return value;
        }
    }
```

Reply:
Same for `SkipMappedValue` class as well